core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -42,20 +41,20 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
       });
   }
 
-  getExecutionStatus(statusCode: number): String {
+  getExecutionStatus(statusCode: number): string[] {
     switch (statusCode) {
       case 0:
-        return ExecutionState.Initializing.toString();
+        return [ExecutionState.Initializing.toString(), "sync", "#a6bd37"];
       case 1:
-        return ExecutionState.Running.toString();
+        return [ExecutionState.Running.toString(), "play-circle", "orange"];
       case 2:
-        return ExecutionState.Paused.toString();
+        return [ExecutionState.Paused.toString(), "pause-circle", "magenta"];
       case 3:
-        return ExecutionState.Completed.toString();
+        return [ExecutionState.Completed.toString(), "check-circle", "green"];
       case 4:
-        return ExecutionState.Aborted.toString();
+        return [ExecutionState.Aborted.toString(), "exclamation-circle", "gray"];
     }
-    return "";
+    return [];
```

Comment:
wouldn't it cause indexoutofbound if the elements are empty? maybe give default values